typed.go

```diff
@@ -41,15 +44,55 @@ encoding definition for standard StarkNet Domain messages
 func (dm Domain) FmtDefinitionEncoding(field string) (fmtEnc []*big.Int) {
 	switch field {
 	case "name":
-		fmtEnc = append(fmtEnc, types.StrToFelt(dm.Name).Big())
+		nameFelt := strToFelt(dm.Name)
+		nameBig, ok := feltToBig(nameFelt)
+		if ok {
+			fmtEnc = append(fmtEnc, nameBig)
+		}
 	case "version":
-		fmtEnc = append(fmtEnc, types.StrToFelt(dm.Version).Big())
+		versionFelt := strToFelt(dm.Version)
+		versionBig, ok := feltToBig(versionFelt)
+		if ok {
+			fmtEnc = append(fmtEnc, versionBig)
+		}
 	case "chainId":
-		fmtEnc = append(fmtEnc, types.StrToFelt(dm.ChainId).Big())
+		chainIdFelt := strToFelt(dm.ChainId)
+		chainIdBig, ok := feltToBig(chainIdFelt)
+		if ok {
+			fmtEnc = append(fmtEnc, chainIdBig)
+		}
 	}
 	return fmtEnc
 }
 
+// strToFelt converts a string containing a decimal, hexadecimal or UTF8 charset into a Felt.
+func strToFelt(str string) *felt.Felt {
+	var f *felt.Felt
```

Comment:
Is it better to have variable f as a named return parameter of the function?

Reply:
From my understanding, it comes down to preference. Naming the return variables can serve as documentation, but it also comes with a risk of accidentally shadowing them. I generally prefer not to name the return variables as I think the code is actually more readable by explicitly declaring them at the top (particularly if you're new to Go etc), you also don't have to worry about shadowing etc.